<launch>
    <include file="$(find course_agv_gazebo)/launch/course_agv_world.launch"/>

	<!-- <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/> -->
	<arg name="map_file" default="$(find course_agv_gazebo)/config/map.yaml"/>
    <!-- Run the map server -->
    <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />

    <param name="robot_name" type="str" value="course_agv" />
    <param name="link_name" type="str" value="robot_chassis" />

    <node pkg="tf" type="static_transform_publisher" name="course_map_tf" args="0 0 0 0 0 0 world_base map 100" />
    <node pkg="course_agv_gazebo" type="robot_tf.py" output="screen" name="robot_pose_tf_publisher"/>

	<node pkg="course_agv_slam" type="icp.py" name="icp" output="screen" launch-prefix="xterm -e">
		
		<!--from a rosbag-->
		<param name="robot_x" value="0"/>
		<param name="robot_y" value="0"/>
		<param name="robot_theta" value="0"/>

		<param name="max_iter" value="10"/>
		<param name="tolerance" value="0"/>
		<param name="dis_th" value="3"/>
	</node>
	
	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find course_agv_slam)/launch/course_agv.rviz" />
	<!-- <node pkg="rviz" type="rviz" name="rviz" args="-d $(find course_agv_gazebo)/launch/course_agv.rviz" /> -->

</launch>
